packages/wallet/src/utils/json-rpc-utils.ts

```diff
@@ -65,3 +71,22 @@ export function createStateFromCreateChannelParams(params: JsonRpcCreateChannelP
     isFinal: false
   };
 }
+
+// TODO: Error handling
+export function createStateFromUpdateChannelParams(
+  state: State,
+  params: JsonRpcUpdateChannelParams
+): State {
+  const {appData, allocations} = params;
+
+  // TODO: check for valid transition using EVM library
+
+  // TODO: Check if this is a final state... I guess it couldn't be
```

Comment:
In the case where the previous `state` `isFinal` I think `UpdateState` should fail, as the exchange of `isFinal` states will be handled by `ConcludeChannel`.

Assuming `s10` is a final state we already have, we can either:
- Have `UpdateChannel` construct `s11` with `isFinal:false` and fail `validTransition` returning an `Invalid Transition Error`
- Have `UpdateChannel` check if `s10` is final and return a bespoke error (Channel being Closed) 

I think we could either:
- cons

Reply:
Yeah good point, I'll add this in another PR